src/gpu/locks.rs

```diff
@@ -13,19 +13,23 @@ fn tmp_path(filename: &str) -> PathBuf {
 
 /// `GPULock` prevents two kernel objects to be instantiated simultaneously.
 #[derive(Debug)]
-pub struct GPULock(File);
+pub struct GPULock {
+    f: File,
```

Comment:
I don't know why it is a struct field now. In case the accessing was the problem, you can access a field in a newstruct via `.0`